bootstrap/kubeadm/internal/controllers/kubeadmconfig_controller.go

```diff
@@ -1026,6 +1104,35 @@ func (r *KubeadmConfigReconciler) reconcileDiscovery(ctx context.Context, cluste
 	return ctrl.Result{}, nil
 }
 
+func (r *KubeadmConfigReconciler) reconcileDiscoveryFile(ctx context.Context, cluster *clusterv1.Cluster, config *bootstrapv1.KubeadmConfig, certificates secret.Certificates) (ctrl.Result, error) {
```

Comment:
nit 
```suggestion
func (r *KubeadmConfigReconciler) reconcileDiscoveryFile(ctx context.Context, cluster *clusterv1.Cluster, cfg *bootstrapv1.FileDiscovery, certificates secret.Certificates) (ctrl.Result, error) {
```

Reply:
`cfg` is used down below as a variable, should I change that to something else?

Reply:
you can drop the variable down below because the func will get in input config.Spec.JoinConfiguration.Discovery.File.KubeConfig as a cfg

Reply:
ah I see what you mean, sounds good